telethon/network/connection/tcpmtproxy.py

```diff
@@ -130,6 +132,21 @@ def address_info(proxy_info):
             raise ValueError("No proxy info specified for MTProxy connection")
         return proxy_info[:2]
 
+    @staticmethod
+    def normilize_secret(secret):
+        if secret[:2] in ["ee","dd"]: # Remove extra bytes
+            secret = secret[2:]
+
+        try:
+            secret_bytes = bytes.fromhex(secret)
+        except:
+            try:
+                secret_bytes = base64.b64decode(secret.encode())
+            except binascii.Error:
+                secret = secret + "=="
```

Comment:
We should just add the padding always instead of guessing (so this `try/except` could then be removed. See https://stackoverflow.com/questions/2941995/python-ignore-incorrect-padding-error-when-base64-decoding.

Reply:
all fixed. pip standard is using Class name for `@staticmethod` instead of `self` but its ok.

Reply:
Might be a habit of mine from adding type hints, where you can't reference the class, because it's definition hasn't been finished yet.

In any case, calling staticmethods via `self` is supported, and allows the class to be renamed without having to modify the calls within itself. I think that's what I've done elsewhere so I'd prefer keeping it consistent.